### PR TITLE
integrate service login

### PR DESCRIPTION
**Pull request by Alejandro-Suarez-Rincon**
# Integrate Service Login

## Description
Este PR tiene la base de la integracion front-back haciendo el inicio se sesion
imagen
imagen

### Summary
Se realizo un cambio en la pagina `/` para que cuando este logeado no muestre los botones de iniciar sesion y registrarse

## Test
Se hizo el test probándolo en el navegador

## Checklist:
- [x]  My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/auth.ts [hunk 1]
```diff
@@ -13,16 +14,32 @@ export const { auth, handlers, signIn, signOut } = NextAuth({
                     .safeParse(credentials);
                 if (parsedCredentials.success) {
                     const { email, password } = parsedCredentials.data;
-                    if (email === "optitech@optitech.com" && password === "optitech")
-                        return {
-                            name: "Optitech",
-                            email: "optitech@gmail.com"
-                        }
+                    console.log(email, password)
```
Comment (JosephSC0121) [inline src/auth.ts:17]: console.log

File: src/app/layout.tsx [hunk 2]
```diff
@@ -19,6 +21,7 @@ export default async function RootLayout({
                     <NextUIProvider>
                         {children}
                     </NextUIProvider>
+                    <ToastContainer />
```
Comment (JosephSC0121) [inline src/app/layout.tsx:24]: qué opinan de esta lib https://sonner.emilkowal.ski/

File: src/modules/auth/services/actions.ts [hunk 3]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+        password: formData.get("password"),
+        passwordReply: formData.get("passwordReply")
+    })
+
+    if (validateFields.success) {
+        if (validateFields.data.password === validateFields.data.passwordReply) {
+            const response = await changePasswordService(validateFields.data.token, validateFields.data.password)
+            if (response) {
+                return {
+                    message: 'La contrasena ha sido actualizada'
```
Comment (JosephSC0121) [inline src/modules/auth/services/actions.ts:83]:
```suggestion
                    message: 'La contraseña ha sido actualizada'
```

File: src/modules/auth/templates/change-password.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                <div>
+                    <h1 className="font-extrabold text-xl">Cambiar Contraseña.</h1>
+                    <p className="text-sm">
+                        Por favor, introduce tu contrasena.
+                    </p>
+                </div>
+                <form action={dispatch} className="flex flex-col gap-4 min-w-80 ">
+                    <div className="mx-5">
+                        <InputPassword
+                            label="Nueva contrasena"
```
Comment (JosephSC0121) [inline src/modules/auth/templates/change-password.tsx:40]:
```suggestion
                            label="Nueva contraseña"
```

File: src/modules/auth/templates/change-password.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                            name="password"
+                            required
+                            radius="sm"
+                            variant="bordered"
+                        />
+                    </div>
+                    <div className="mx-5">
+                        <InputPassword
+                            label="Repetir contrasena"
```
Comment (JosephSC0121) [inline src/modules/auth/templates/change-password.tsx:49]:
```suggestion
                            label="Repetir contraseña"
```

File: src/modules/auth/templates/principal.tsx [hunk 6]
```diff
@@ -1,29 +1,49 @@
...
+import { Button } from "@/modules/common/components/button";
+
+export type PrincipalProps = {
+  // TODO: Add type to clientInfo
+  clientInfo: any;
+};
 
-export default function Principal() {
+export default function Principal(props: PrincipalProps) {
+  console.log(props.clientInfo);
```
Comment (JosephSC0121) [inline src/modules/auth/templates/principal.tsx:12]: console.log

File: src/app/layout.tsx (hunk 2 above)
Comment (CamiloArias12) [reply src/app/layout.tsx:24]: si esta mejor de configurar

Comment (CamiloArias12) [reply src/app/layout.tsx:24]: [1ca7356](https://github.com/OwlByTech/optitech-web/pull/7/commits/1ca7356fe550ca83ec147367936c9763bcf2d6e6)

**Review by CamiloArias12 — APPROVED** [review]

File: src/modules/auth/templates/principal.tsx (hunk 6 above)
Comment (Alejandro-Suarez-Rincon) [reply src/modules/auth/templates/principal.tsx]: https://github.com/OwlByTech/optitech-web/pull/12/commits/2f973f72f860017f28ebc354a3eb7164d1b70a1b

File: src/auth.ts (hunk 1 above)
Comment (Alejandro-Suarez-Rincon) [reply src/auth.ts]: https://github.com/OwlByTech/optitech-web/pull/12/commits/2f973f72f860017f28ebc354a3eb7164d1b70a1b

**Review by JosephSC0121 — APPROVED** [review]

**Pull request merged**